enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,901 @@
+                "kind": "EtcdBackup", 
+                "version": "v1beta2"
+            }
+        },
+        {
+            "name":"olm.channel",
+            "value": "singlenamespace-alpha"
+        }
+    ],
+    "relatedImages" : [
+        {
+            "name": "etcdv0.9.0",
+            "image": "quay.io/coreos/etcd-operator@sha256:db563baa8194fcfe39d1df744ed70024b0f1f9e9b55b5923c2f3a413c44dc6b8"
+        }
+    ]
+}
+```
+
+### Using the package json representation
+
+Once every package inside an index is represented using a json/yaml config file, the config files can be leveraged to inspect/update/validate the packages inside the index without needing to rebuild the index from scratch.  
+#### Inspecting the packages inside an index
+
+A new sub-command `inspect` will be introduced under `opm index`.
+
+When `opm index inspect` is summoned, the index image will be downloaded, and the json representations of the packages inside the index will be unfurled in a folder with the same name as the index. 
+
+```bash 
+$ opm index inspect --index=docker.io/my-namespace/community-operators --output=json
+$ cd community-operators
+$ tree
+.
+├── etcd.json
+├── amqstreams.json
+```
+#### Editing a package inside an index
+
+A new sub-command `update` will be introduced under `opm index`, which will take a folder containing json representations of packages as it's input along with an existing index image tag, and will replace the edited config files in the index container. 
+
+```bash
+$ opm index inspect --index=quay.io/some-namespace/my-index 
+$ // edit a my-index/package-config.json 
+$ opm index update my-index --tag=docker.io/some-namespace/my-index:latest
+``` 
+
+> Note: For a seamless UX, the following things should be consider during implementation when the `opm index update` command is summoned: 
+> 1. Validate the files inside the folder given as input to the command to ensure they are valid json representation of packages (and not random files with random content, which could be potential security risks) 
+> 2. If the files are valid, take a git diff between the present content of the index (i.e the existing json representations) and the content of the folder given as input. Display the diff as an output of the command with the following message 
+> "The following changes will be updated in the index:"
+> If there is no diff between the content in the container image and the content of the local folder, throw an error with the message "No diff found between index and local content to update."    
+
+
+#### Creating new indexes using the config files from an existing index
+
+A new sub-command `create` will be introduced under `opm index`, that will take a folder containing json/yaml representation of packages, and a container image tag as input. The sub-command will create a new index container, walk the tree of files to find the json representations and store them in the container. 
```

Comment:
I may need to re-read this enhancement again, but in the meantime, is this command something we can expect will be called under-the-hood to generate the overall config JSON/YAML file that we copy to the index image container image?

Reply:
yes